python/smashbox/utilities/__init__.py

```diff
@@ -346,28 +360,67 @@ def run_ocsync(local_folder, remote_folder="", n=None, user_num=None):
         logger.info('sync finished: %s',datetime.datetime.now()-t0)
         ocsync_cnt[current_step]+=1
 
+def _prop_check(path,user_num=None,depth="0"):
+    """ Private function to implement other utilities.
+    """
+    import smashbox.curl
+    c = smashbox.curl.Client()
+    url = oc_webdav_url(remote_folder=path, user_num=user_num)
+    query = """<?xml version="1.0" encoding="UTF-8"?><D:propfind xmlns:D="DAV:"><D:allprop/></D:propfind>"""
+
+    r = c.PROPFIND(url,query,depth=depth)
+    return r
+
+
 
 def webdav_propfind_ls(path, user_num=None):
-    runcmd('curl -s -k %s -XPROPFIND %s | xmllint --format -'%(config.get('curl_opts',''),oc_webdav_url(remote_folder=path, user_num=user_num)))
+    import xml.dom.minidom
+    r = _prop_check(path,user_num,depth="1")
+    x=xml.dom.minidom.parseString(r.response_body)
+    logger.info('Listing %s by PROPFIND: %s'%(repr(path),x.toprettyxml()))
 
 def expect_webdav_does_not_exist(path, user_num=None):
-    exitcode,stdout,stderr = runcmd('curl -s -k %s -XPROPFIND %s | xmllint --format - | grep NotFound | wc -l'%(config.get('curl_opts',''),oc_webdav_url(remote_folder=path, user_num=user_num)))
-    not_exists = stdout.rstrip() == "1"
-    error_check(not_exists, "Remote path does not %s exist but should" % path)
+
+    r = _prop_check(path,user_num)
+    error_check(r.rc >= 400,"Remote path exists: %s" % path) # class 4xx response is OK
```

Comment:
I think this should check for 404, to avoid problems with missing authentication when someone implements a new test.